resolve/utilities/version.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef RESOLVE_VERSION_H
+#define RESOLVE_VERSION_H
+
+
+/**
+ * Sets major, minor, and patch versions for current ReSolve build. The user is
+ * responsible for free'ing this memory.
+ */
+int ResolveVersionGetVersion(int *, int *, int *);
```

Comment:
Some suggested style formatting and comment corrections:
```suggestion
/**
 * Sets major, minor, and patch versions for current ReSolve build. The user is
 * responsible for ensuring `int`s are allocated.
 */
int ResolveVersionGetVersion(int*, int*, int*);
```